src/hexEditorRegistry.ts

```diff
@@ -68,6 +70,20 @@ export class HexEditorRegistry extends Disposable {
 		};
 	}
 
+	/** adds a diff model */
+	public addDiff(diffModelBuilder: HexDiffModelBuilder) {
```

Comment:
The lifecycle of this is kind of weird and I'm not sure all cases are handled, such as reopening an editor without re-running the "compare selected". It seems like the main point of this is sharing the model builder between the two URIs.

Instead I would suggest stashing some query parameters when you build the URIs like `?token={uuid}&side={modified|original}` having the getDiff be something like

```js
getDiff(uri: vscode.Uri): { builder: HexDiffModelBuilder, dispose: () => void } {
    const { token, side } = parseQueryStringFrom(uri);
    if (!this.diffsBuilder.get(token)) {
        this.diffsBuilder.set(token, { refCount: 0, value: new DiffsBuilder() });
    }
    const builder = this.diffsBuilder.get(token);
    builder.refCount++;
    builder.value.set(side, uri);

    return { builder: builder.value, dispose: () => { /* refCount--, remove if 0 */ }}
}
```

...and have `HexDiffModelBuilder.build()` await both sides being set similarly to how it works for the documents today. That approach should allow things to work and the editor to be reopened/moved/duplicated without issue.

Reply:
Thanks for the great suggestion, it made the lifecycle a lot simpler! I did some small tweaks, but overall it should be the same.